tests/periph_spi/tests/04__periph_spi_timer.robot

```diff
@@ -13,66 +13,88 @@ Test Teardown       Run Keywords    SPI Release Should Succeed
-    ${sys_clk} =  Set Variable          ${RESULT['data']}
-    API Call Should Succeed             PHiLIP.Read Reg         spi.byte_ticks
-    ${spi_speed_limits} =  Set Variable  
-    ...  ${spi_speed_limits(${clk_speed_value}, ${sys_clk}, ${size})}
-    Should Be True  
-    ...  ${spi_speed_limits}[0] < ${${RESULT['data']} * ${size}} and ${spi_speed_limits}[1] > ${${RESULT['data']} * ${size}}
-    ...  Time should be between ${convert_ticks_to_us(${spi_speed_limits[0]},${sys_clk})}us and ${convert_ticks_to_us(${spi_speed_limits[1]},${sys_clk})}us but was ${convert_ticks_to_us(${RESULT['data']} * ${size}, ${sys_clk})}us
+    SPI Acquire Should Succeed          0                   ${clk_speed_string}
+    SPI Transfer Bytes Should Succeed   cont=0              in_len=${size}
+    API Call Should Succeed             PHiLIP.Read Reg     sys.sys_clk
+    Set Test Variable                   ${sys_clk}          ${RESULT['data']}
+    API Call Should Succeed             PHiLIP.Read Reg     spi.byte_ticks
+    Set Test Variable                   ${byte_ticks}       ${RESULT['data']}
+    Set Test Variable                   ${spi_speed_limits}
+    ...                                   ${spi_speed_limits(${clk_speed_value}, ${sys_clk}, ${size})}
+    Set Test Variable                   ${lower_time_limit}
+    ...                                   ${ticks_to_us(${spi_speed_limits[0]}, ${sys_clk})}
+    Set Test Variable                   ${upper_time_limit}
+    ...                                   ${ticks_to_us(${spi_speed_limits[1]}, ${sys_clk})}
+    Set Test Variable                   ${measured_time}
+    ...                                   ${ticks_to_us((${byte_ticks} * ${size}), ${sys_clk})}
+    Run Keyword And Warn On Failure     Compare Time Measures
+
+    Record Property  Measured time ${clk_speed_string}Hz ${size}Byte  ${measured_time}us
+    Record Property  Lower limit ${clk_speed_string}Hz ${size}Byte    ${lower_time_limit}us
+    Record Property  Upper limit ${clk_speed_string}Hz ${size}Byte    ${upper_time_limit}us
+
+
+Compare Time Measures
+    [Documentation]     This is part of the main routine. Here the measured times
+    ...                 are compared to the allowed time window.
+    ...                 Seperating it enables to execute this part with Warn On Failure,
+    ...                 while keeping the possibility of receiving a proper failure
+    ...                 in case an error occured.
+    Should Be True
+    ...  ${spi_speed_limits}[0] < (${byte_ticks} * ${size}) and ${spi_speed_limits}[1] > (${byte_ticks} * ${size})
+    ...  Time should be between ${lower_time_limit}us and ${upper_time_limit}us but was ${measured_time}us
 
 
 *** Test Cases ***
-Clock Speed 100k Should Succeed
+Clock Speed 100k Should Succeed 2 Byte
     [Documentation]  Checks the clock speed for 100kHz
-    SPI Clock Speed Check    100k     100000    1
+    SPI Clock Speed Check Setup    100k     100000    2
```

Comment:
We can probably keep this as 1 when it is that slow.  It may also be good to add backoff to the philip numbers to help with accuracy.